app/auth.py

```diff
@@ -0,0 +1,146 @@
+from __future__ import annotations
+
+import json
+import logging
+
+from http import HTTPStatus
+from fastapi import APIRouter, Depends, Form
+from fastapi.responses import RedirectResponse
+from typing import Annotated
+
+from auth_fastapi import (
+    make_email_password,
+    email_password as core_email_password,
+)
+
+from .config import BASE_URL
+from .edgedb_client import client
+from .queries import create_user_async_edgeql as create_user_qry
+
+logger = logging.getLogger("fast_jelly")
+router = APIRouter()
+email_password = make_email_password(
+    client,
+    verify_url=f"{BASE_URL}/auth/verify",
+    reset_url=f"{BASE_URL}/ui/reset-password",
+)
+
+
+@router.post(
+    "/auth/register",
+    response_class=RedirectResponse,
+    status_code=HTTPStatus.SEE_OTHER,
+)
+async def register(
+    email: Annotated[str, Form()],
+    sign_up_response: Annotated[
+        core_email_password.SignUpResponse, Depends(email_password.handle_sign_up)
+    ],
+):
+    if not isinstance(sign_up_response, core_email_password.SignUpFailedResponse):
+        user = await create_user_qry.create_user(
+            client,
+            name=email,
+            identity_id=sign_up_response.identity_id,
+        )
+        print(f"Created user: {json.dumps(user, default=str)}")
+
+    match sign_up_response:
+        case core_email_password.SignUpCompleteResponse():
+            return "/"
+        case core_email_password.SignUpVerificationRequiredResponse():
+            return "/signin?incomplete=verification_required"
+        case core_email_password.SignUpFailedResponse():
+            logger.error(f"Sign up failed: {sign_up_response}")
+            return "/signin?error=failure"
+        case _:
+            raise Exception("Invalid sign up response")
+
+
+@router.post(
+    "/auth/authenticate",
+    response_class=RedirectResponse,
+    status_code=HTTPStatus.SEE_OTHER,
+)
+async def authenticate(
+    sign_in_response: Annotated[
+        core_email_password.SignInResponse, Depends(email_password.handle_sign_in)
+    ],
+):
```

Comment:
Why not annotate the input with `SignInBody` here? The generated OpenAPI docs could benefit from such typing.